coderd/templateversions.go

```diff
@@ -1306,6 +1306,7 @@ func (api *API) postTemplateVersionsByOrganization(rw http.ResponseWriter, r *ht
 			CreatedAt:      database.Now(),
 			UpdatedAt:      database.Now(),
 			Name:           req.Name,
+			Message:        req.Message,
```

Comment:
We do so in the CLI currently, the CLI says "Uploaded from the CLI". I don't think there's anything meaningful the API could mention here? Or do you have any ideas?

Reply:
What about the template editor? I suppose we'd pre-fill something?

Reply:
I think an empty message is fine and let the viewer add a default if there is one.

Like if you want `no message provided`, the UI can just display that if it's empty.

Reply:
I'm also in favor of allowing an empty message. The question is only, do we push users to add a message (bypass via `--allow-empty`), or do we default to empty with no notice.

If prompting is enabled, we can ask for the message if it's missing. But perhaps `--yes` should fail if a message is empty and there's no `--allow-empty`. Otherwise I feel users will forget to add the message more often than not.

Ref: https://github.com/coder/coder/pull/8336#discussion_r1255481350

Reply:
We've settled on not enforcing a message for now and potentially allow updating the message when promoting a version.